### DISCUSSION TITLE
Does the weather impact the readings from the sensors?

### DISCUSSION THREAD

**Discussion by saksham2001** [carla-simulator/carla#6757] [Q&A]
Different weather conditions like fog, snowing, rain, etc are known to cause perturbations in readings of many sensors especially sensors like LiDAR. How well are these represented in CARLA? I know that we can control many parameters for the LiDAR including the atmospheric attenuation, but I want to use multiple sensors (RGB camera and LiDAR) and want to ensure that the artefacts visible in the point cloud form the LiDAR correlate with what the RBG camera can see.

### Effect of Snowing [1]
<img width="600" alt="Screenshot 2023-09-06 at 6 39 23 PM" src="https://github.com/carla-simulator/carla/assets/44300901/f3eafb6f-09e3-43fa-82fd-b422598a03d3">

1. Hahner, M., Sakaridis, C., Bijelic, M., Heide, F., Yu, F., Dai, D., & Van Gool, L. (2022). Lidar snowfall simulation for robust 3d object detection. In Proceedings of the IEEE/CVF Conference on Computer Vision and Pattern Recognition (pp. 16364-16374).

**Comment by aryuemaan** [comment]
In the simulator it show's 50 % accurate for any sensor and yes the results might have differences.

**Comment by deyang2000** [comment]
I discovered that Carla only offers simulations for rainy and foggy weather conditions (although the windy simulation is not useful for my purposes). Therefore, I utilize Martin Hahner's snowfall simulation to create a snowy weather scenario.

**Reply by Lizhe1228** [reply]
How did you create the snowy environment? Can you give me some guidance and suggestions? I am also interested in the snowy environment in Carla.

**Comment by saksham2001** [comment]
Thank you for the reply, @deyang2000. I have used Martin Hahner's snowfall simulation; it's excellent if only using LiDAR. However, I have not found a method to do that consistently for multiple modalities (like camera and LiDAR).

**Reply by deyang2000** [reply]
I asked my teammates and got some information, mostly about a senior thesis (studying 3D object detection in adverse weather). The authors use different methods to simulate RGB and point clouds (the origin RGB and point clouds come from KITTI,sunny day).

In this paper he said:

![snowsim](https://github.com/carla-simulator/carla/assets/129513825/efe41d33-eff7-430f-86b6-58e8582ce1b5)
This paper finds that there are many methods to realize the effect rendering of image-based fog and snow, and finally chooses imgaug third-party library in python environment for rendering, because this method has the following advantages: First, the method can render a variety of adverse weather effects not limited to fog and snow based on the image, and has a high degree of rendering freedom for each weather type, and further supports the free combination of different weather effects. Secondly, the rendering effect of this method is closer to the actual situation. Using the bad weather data rendered based on this method as training data can improve the robustness of the model against real bad weather scenes. Finally, the method is not difficult to operate and occupies less computing resources, which is conducive to the efficient rendering of mass images in bad weather. In order to render different influence degree of snow in the image effect, this article will 𝑙 𝑖 𝑔 ℎ 𝑡 𝑛 𝑒 𝑠 𝑠 _ 𝑡 ℎ 𝑟 𝑒 𝑠 ℎ 𝑜 𝑙 𝑑 parameter is set to (100, 255), will 𝑙 𝑖 𝑔 ℎ 𝑡 𝑛 𝑒 𝑠 𝑠 _ 𝑚 𝑢 𝑙 𝑡 𝑖 𝑝 𝑙 𝑖 𝑒 𝑟 parameter is set to (1.24, 1.48), the ground snow rendering unchanged, Through adjusting 𝑓 𝑙 𝑎 𝑘 𝑒 _ 𝑠 𝑖 𝑧 𝑒 parameters and 𝑠 𝑝 𝑒 𝑒 𝑑 parameters will be rendered into light, moderate snow, snow, snow effect specifically: When 𝑓 𝑙 𝑎 𝑘 𝑒 _ 𝑠 𝑖 𝑧 𝑒 parameter is set to (0.1, 0.2), 𝑠 𝑝 𝑒 𝑒 𝑑 parameter is set to (0.004, 0.006), this article will apply colours to a drawing of the snow is defined as a light snow; When 𝑓 𝑙 𝑎 𝑘 𝑒 _ 𝑠 𝑖 𝑧 𝑒 parameter is set to (0.3, 0.4), 𝑠 𝑝 𝑒 𝑒 𝑑 parameter is set to (0.006, 0.008), the article will apply colours to a drawing of snow defined as moderate snow; When 𝑓 𝑙 𝑎 𝑘 𝑒 _ 𝑠 𝑖 𝑧 𝑒 parameter is set to (0.5, 0.6), 𝑠 𝑝 𝑒 𝑒 𝑑 parameter is set to (0.008, 0.01), this article will apply colours to a drawing of snow is defined as the snow.

The fig is his snow sim work.
(For reference only, in order to respect the intellectual property rights of the author, please do not use it elsewhere)

While he solved some of the problems in the snow simulation, I would say that there is very little relationship between point clouds and RGB because he generated point clouds and RGB using completely different methods.
Of course, this is only a master's thesis, in order to complete the graduation task, the logic is not very rigorous.

It might help you.
